ministryofjustice/laa-review-criminal-legal-aid
# CRIMRE-374 address error handling, pages and reporting

## Description of change

1. Standardise error responses to reduce information leakages:

- When not authenticated, all requests to service and admin URL redirect to sign in regardless of the existing of the resource.
- When authenticated admin, all requests to the service redirect to the user admin
- When authenticated service user, all requests to user admin render 404

2. Simplify all but application_not_found error page to reduce risk of error loops.

3. Ensure error pages do not render flash messages by default.

4.  User errors controller to render all application and middleware errors.

## Link to relevant ticket
[CRIMRE-374](https://dsdmoj.atlassian.net/browse/CRIMRE-374)

## Notes for reviewer

## Screenshots of changes (if applicable)

### Before changes:

### After changes:

## How to manually test the feature

See ticket for criteria. 

To render the error pages in development for testing add the following line to config/environments/development.rb 

```
config.action_dispatch.show_exceptions = true
config.consider_all_requests_local = false
```


[CRIMRE-374]: https://dsdmoj.atlassian.net/browse/CRIMRE-374?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
